src/SFML/Graphics/Sprite.cpp

```diff
@@ -72,17 +68,15 @@ void Sprite::setTextureRect(const IntRect& rectangle)
     if (rectangle != m_textureRect)
     {
         m_textureRect = rectangle;
-        updatePositions();
-        updateTexCoords();
+        updateVertices();
     }
 }
 
 
 ////////////////////////////////////////////////////////////
 void Sprite::setColor(const Color& color)
 {
-    // Update the vertices' color
-    for (auto& vertex : m_vertices)
+    for (Vertex& vertex : m_vertices)
```

Comment:
It absolutely is a simplification, as we were using a more powerful tool (type inference) in a situation where a more limited one (simple concrete type) suffices -- it's not about "style" at all, "style" is picking Allman brace style over K&R brace style. 

Type deduction is not necessary here, nor it reduces repetition, nor it increases readability. All it does is obfuscate the type of `vertex`. As you reminded me of [this principle yourself](https://github.com/SFML/SFML/pull/3061#issuecomment-2147960874), now it is my turn to remind you of the same philosophy.

Opening up an issue to discuss what has been [already discussed and decided upon](https://github.com/SFML/SFML/pull/3061#issuecomment-2147960874)...

> type inference -- `auto` and `decltype` should only be used where they considerably increase readability (e.g. iterators), lack of types for basic variables can make code harder to understand

...is a complete waste of time, and this PR is perhaps being "slowed down" only because you brought up a concern about a valid removal of `auto` that is in line with our Modern C++ feature usage policy.

Reply:
I agree with Chris and basically disagree with every points you make Vittorio, but I also fall into the [AAA (almost always auto) camp](https://duerrenberger.dev/blog/2021/11/19/almost-always-auto/), so I'm certainly very bias on the topic.

> It absolutely is a simplification, as we were using a more powerful tool (type inference) in a situation where a more limited one (simple concrete type) suffices -- it's not about "style" at all, "style" is picking Allman brace style over K&R brace style.

In the most obtuse, pedantic sense you can say yes it's not a "style" choice, but it falls into the same category. The change has no effect on the outcome as the compiler will do the work for you.

Either way it's a "stylistic", personal choice of how to use the language feature and there's little sense in trying to make technical arguments about it.

> Opening up an issue to discuss what has been [already discussed and decided upon](https://github.com/SFML/SFML/pull/3061#issuecomment-2147960874)...
> 
> > type inference -- `auto` and `decltype` should only be used where they considerably increase readability (e.g. iterators), lack of types for basic variables can make code harder to understand

I've no idea where that quote is from, but the linked discussion didn't talk about `auto` and `decltype` and certainly didn't decide or set on any policy.

> ...is a complete waste of time, and this PR is perhaps being "slowed down" only because you brought up a concern about a valid removal of `auto` that is in line with our Modern C++ feature usage policy.

Let's maybe not start with such blaming games. We should be open for any feedback, even if you don't like the topic.

Reply:
> I agree with Chris and basically disagree with every points you make Vittorio

Of course...

> but I also fall into the [AAA (almost always auto) camp](https://duerrenberger.dev/blog/2021/11/19/almost-always-auto/), so I'm certainly very "bias" on the topic (i.e. holding a personal stylistic opinion).

Then you misundestand what AAA actually means.

The stylistical choice is:

```cpp
auto i0 = int{0};
int i1 = 0;
```

Those two are equivalent and only differ in style. `i0` is an example of AAA. 

It is absolutely not the same as the comparison betwen:

```cpp
auto i0 = 0;
int i1 = 0;
```

Because now `i0`'s type is not visible anymore and we are relying on type inference.

AAA does not imply type inference.

> > It absolutely is a simplification, as we were using a more powerful tool (type inference) in a situation where a more limited one (simple concrete type) suffices -- it's not about "style" at all, "style" is picking Allman brace style over K&R brace style.
> 
> In the most obtuse, pedantic sense you can say yes it's not a "style" choice, but it falls into the same category. The change has no effect on the outcome as the compiler will do the work for you.

Implying that any change that doesn't have an impact on the final generated binary is stylistical is wrong and dangerous.

We could use `auto&&` in most places where `auto` suffices without any change to the program's behavior. Is it style?

We could use `auto` in most places where `T` suffices. Is it style? 

> Either way it's a "stylistic", personal choice of how to use the language feature and there's little sense in trying to make technical arguments about it.

Absolutely not, there are real readability and usability impacts when using type inference over a concrete type, sometimes positively and sometimes negatively.

Technical arguments on the use of type inference have been made ever since `auto` was introduced and for good reason, as it is a powerful feature whose use should be judicious.

> > Opening up an issue to discuss what has been [already discussed and decided upon](https://github.com/SFML/SFML/pull/3061#issuecomment-2147960874)...
> > > type inference -- `auto` and `decltype` should only be used where they considerably increase readability (e.g. iterators), lack of types for basic variables can make code harder to understand
> 
> I've no idea where that quote is from, but the linked discussion didn't talk about `auto` and `decltype` and certainly didn't decide or set on any policy on how to use and when to use `auto`/type inference.

I meant to link to this. https://github.com/SFML/SFML/wiki/SD:-CPP-17-Support#implementation-focused

A page *you* created and revised. 😅

> > ...is a complete waste of time, and this PR is perhaps being "slowed down" only because you brought up a concern about a valid removal of `auto` that is in line with our Modern C++ feature usage policy.
> 
> Let's maybe not start with such blaming games. We should be open for any feedback, even if you don't like the topic.

The feedback could have been delivered without accusing my change of being stylistical (it isn't) and implying that I was wasting the reviewer's time.

Had the initial feedback been delivered purely on a technical basis I would have not responded in tone.